src/lib.rs

```diff
@@ -1384,8 +1385,8 @@ pub enum RequestTokenError<T: ErrorResponseType + Send + Sync + 'static> {
     /// Failed to parse server response. Parse errors may occur while parsing either successful
     /// or error responses.
     ///
-    #[fail(display = "Failed to parse server response")]
-    Parse(#[cause] serde_json::error::Error),
+    #[fail(display = "Failed to parse server response (response: {})", _1)]
+    Parse(#[cause] serde_json::error::Error, String),
```

Comment:
Let's return `Vec<u8>` instead of `String` to support the non-UTF8 case

Reply:
This will make it hard to debug, but at least doable. I'll take it.